### PR TITLE
Improve Travis 

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by ejgallego** [comment]
I'll need a couple of days to properly review this.

**Comment by SkySkimmer** [comment]
@ejgallego bump

**Comment by ejgallego** [comment]
Gonna take a while sorry, but it is on my TODO list so don't worry.

File: .travis.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ env:
   - TEST_TARGET="ci-fiat-crypto"
   - TEST_TARGET="ci-fiat-parsers"
   - TEST_TARGET="ci-flocq"
-  - TEST_TARGET="ci-hott"
+  - TEST_TARGET="ci-hott" BUILD_TARGET="tools coqbinaries pluginsopt states"
```
Comment (ejgallego) [inline .travis.yml]: I don't favor this kind of customization as for a) IMHO it should be contained in the HoTT script overlay; b) we will share the same Coq build anyways soon thanks to Travis build stages.

File: Makefile.dev [hunk 2]
```diff
@@ -97,7 +97,10 @@ minibyte: $(COQTOPBYTE) pluginsbyte
 pluginsopt: $(PLUGINSOPT)
 pluginsbyte: $(PLUGINS)
 
-.PHONY: coqlight states miniopt minibyte pluginsopt pluginsbyte
+# This should build all the ocaml code but not (most of) the .v files
+coqocaml: tools coqbinaries pluginsopt coqide
```
Comment (ejgallego) [inline Makefile.dev]: Isn't ide-toploop missing here too?

**Review by ejgallego — APPROVED** [review]
Aside from the minor comment the PR looks great to me.

If you could please add the two commits in #629 too it would be great, otherwise I'll rebase after this gets merged.

File: .travis.yml (hunk 1 above)
Comment (SkySkimmer) [inline .travis.yml]: Build stages are definitely a better solution, they just didn't exist when I made the commit ;)

**Comment by SkySkimmer** [comment]
#629 included.

I forgot to build the debug printers in `coqocaml`, this has been fixed.

**Comment by ejgallego** [comment]
Thanks!

**Comment by SkySkimmer** [comment]
Had to rebase since fiat-parsers uses recently merged transparent abstract.

**Comment by SkySkimmer** [comment]
*facepalm* of course Coq's travis merges before testing so I didn't need to do that. Oh well.

**Comment by maximedenes** [comment]
@ejgallego Does this now seem ready to merge?

**Comment by ejgallego** [comment]
Yup, I said so and updated accordingly the flag.

Note that is changing the warnings policy, now Travis will fail for non-properly taken care of warnings. I don't particularly care, other than the pain of having to pass -warn-on-error every time to configure...

**Comment by maximedenes** [comment]
@SkySkimmer Could you rebase to clean a bit the history now that #629 was merged? There will be no conflicts. Thanks!

**Comment by SkySkimmer** [comment]
Rebased.

**Pull request merged**